main.cpp

```diff
@@ -0,0 +1,225 @@
+*   Including: Greatest Common Denominator
+               Fibonacci Numbers
+               Integers to Roman Numerals
+               Integers to Alphabetized Integers
+               Magic Number Riddle
+*/
+
+#include <iostream>
+#include <cmath>
+#include <string>
+
+using std::cout;
+using std::endl;
+using std::abs;
+using std::string;
+
+int gcd( int x, int y );
+int gcd_iter( int x , int y );
+int fib( int f , int f1 = 1, int f2 = 1);
+int fib_iter( int f );
+string int_to_roman( int r , string rn = "");
+string int_to_word( int w, string nw = "");
+int magic( int m );
+
+int main()
+{
+    cout << gcd(24,600) << endl;
+    cout << gcd_iter(24,600) << endl;
+    cout << fib(11) << endl;
+    cout << fib_iter(11) << endl;
+    cout << int_to_word(18818) << endl;
+    magic(12);
+
+    return 0;
+}
+
+int gcd( int x, int y )
+{
+    x = abs(x);
+    y = abs(y);
+
+    if( x == 0 || y == 0 )
+        return x+y;
+
+    if( x > y )
+        return gcd( x-y , y);
+    else
+        return gcd( x , y-x );
+}
+
+int gcd_iter( int x , int y )
+{
+    x = abs(x);
+    y = abs(y);
+
+    while(x && y)
+        if( x > y )
+            x -= y;
+        else
+            y -= x;
+
+    return x+y;
+}
+
+int fib( int f , int f1, int f2 )
+{
+    if( f <= 2 )
+        return f2;
+    else
+    {
+        int f3;
+        f3 = f1 + f2;
+        f1 = f2;
+        f2 = f3;
+        f--;
+        if( f != 0)
+            return fib( f , f1 , f2 );
+        else
+            return f3;
+    }
+}
+
+int fib_iter( int f )
+{
+    int f1 = 1;
+    int f2 = 1;
+    int f3;
+    int n = 3;
+
+    if( f <= 2)
+        return f2;
+    else
+        while( n < (f + 1))
+        {
+            f3 = f1 + f2;
+            f1 = f2;
+            f2 = f3;
+            n++;
+        }
+    return f3;
+}
+
+string int_to_roman( int r , string rn)
+{
+    if( r > 4000)
+        return "Out of bounds";
+    else if( r >= 1000 )
+        return int_to_roman( r - 1000, rn + "M" );
+    else if( r  >= 900 )
+        return int_to_roman( r - 900, rn + "CM" );
+    else if( r >= 500 )
+        return int_to_roman( r - 500, rn + "D" );
+    else if( r >= 400 )
+        return int_to_roman( r - 500, rn + "CD" );
+    else if( r >= 100 )
+        return int_to_roman( r - 100, rn + "C" );
+    else if( r >= 90 )
+        return int_to_roman( r - 90, rn + "XC" );
+    else if( r >= 50 )
+        return int_to_roman( r - 50, rn + "L" );
+    else if( r >= 40 )
+        return int_to_roman( r - 40, rn + "XL" );
+    else if( r >= 10 )
+        return int_to_roman( r - 10, rn + "X" );
+    else if( r >= 9 )
+        return int_to_roman( r - 9, rn + "IX" );
+    else if( r >= 5 )
+        return int_to_roman( r - 5, rn + "V" );
+    else if( r >= 4 )
+        return int_to_roman( r - 4, rn + "IV" );
+    else if ( r >= 1 )
+        return int_to_roman( r - 1, rn + "I" );
+    else
+        return rn;
+}
+
+string int_to_word( int w, string nw )
```

Comment:
could be further generalized and condensed -- but overall a really good solution